src/main/java/fi/aalto/cs/apluscourses/ui/base/TextField.java

```diff
@@ -30,6 +36,11 @@ public void removeUpdate(DocumentEvent documentEvent) {
 
       @Override
       public void changedUpdate(DocumentEvent documentEvent) {
+        try {
+          throw new OperationNotSupportedException();
+        } catch (OperationNotSupportedException e) {
+          logger.info("Operation not supported", e);
+        }
```

Comment:
I wouldn't throw something just to catch it immediately, so I would either keep just the throw itself or the info log

Reply:
Oh, this is a similar issue I commented on elsewhere. It cannot throw because this method is called when that event triggers. It should do nothing and return normally and quietly.